internal/util/exit.go

```diff
@@ -1,5 +1,29 @@
 package util
 
-import "os"
+import (
+	"os"
 
-var Exit = os.Exit
+	"github.com/edulinq/autograder/internal/log"
+)
+
+var (
+	ShouldExit   = true
+	LastExitCode = 0
+)
+
+func Exit(code int) {
+	SetExitCode(code)
+
+	if ShouldExit {
+		log.Debug("Exiting with code.", log.NewAttr("code", code))
```

Comment:
You can just say "Exiting" (since all exits always have a code (even if it is zero)).